src/screen.js

```diff
@@ -0,0 +1,65 @@
+
+import config from './config'
+
+const AlpineScreenMagicMethod = {
+    start() {
+        let initialized
+
+        Alpine.addMagicProperty('screen', function ($el) {
+            // bind resize event to window with debounce
+            let update
+            const updateScreen = () => {
+                clearTimeout(update)
+                update = setTimeout(() => {
+                    $el.__x.updateElements($el)
+                }, 150)
+            }
+
+            // bind resize event to window if not initialized
+            if (!initialized) {
+                window.addEventListener('resize', updateScreen)
+            }
+
+            // set initialized to prevent multiple calls
+            initialized = true
+
+            return function (target) {
+                // Get current window dimensions
+                const width = window.innerWidth
+                const height = window.innerHeight
+
+                // If size provided as number, early return
+                if (Number.isInteger(target)) return target <= width
+
+                // Breakpoints and extras
+                const screen = {
+                    ...config.get('breakpoints'),
+                    touch: 'ontouchstart' in window,
+                    portrait: height > width,
```

Comment:
It is screen related. It is another way of checking if screen is on mobile device. Your perspective is probably from breakpoints so this not make sense to you a lot :D

Reply:
Not only mobile devices support `touch` though. Screen sizes and position make sense but this doesn't seem to fit. 

Plus functionality wise it doesn't match with everything else here. Widths and orientations can change but `touch` support is a constant on the device.

Reply:
```html
<div x-show="$screen('portrait'), $screen('lg') && $screen('touch')> 
   I need to be visible if screen is desktop size and support touch
</div>
```
Is this not make sense to you?

Consider uncommon screens. (I have worked on many different screen devices, there are weird cases where device orientation, size, etc. is not common)

Reply:
I know _why_ you want it but I just don't think it fits here. You're checking whether a feature is supported instead of the current state of the browser width. Anyway, I don't necessarily feel strongly against it. Lets see what the others say.

Reply:
Yeah sure, i am not strongly protecting it as well. I just think it's useful where it can detect it. Also IMO $screen helper should provide anything related to screen. 

For your reference [check this](https://www.npmjs.com/package/vue-screen#as-computed-properties)

Reply:
That plugin isn't very popular (relative to other Vue plugins) so I don't think it's something to consider an authority on the topic.

But I think if you wanted to include all screen-related properties then I think that would work. Here's some more: https://developer.mozilla.org/en-US/docs/Web/API/Screen

Reply:
I like the idea of a support helper if there are multiple things to detect. E.g. You'll show a button only if geolocation is supported.

Other thing to investigate if we support orientation, does resize catches when you rotate a mobile phone or do we need a separate listener for it? (we don't need to find out today, i just posted the question so we won't forget :D ).

Reply:
This does feel better:

```html
<div x-show="$screen('portrait') && $screen('lg') && $supports('touch')> 
   I need to be visible if screen is desktop size, portrait and supports touch
</div>
```

Just noticed your comment in the div there even says "supports", so you subconsciously agree :)

> Other thing to investigate if we support orientation, does resize catches when you rotate a mobile phone or do we need a separate listener for it? (we don't need to find out today, i just posted the question so we won't forget :D ).

I think this was an issue about 5 years ago but last I checked it was okay now and worked as expected and they deprecated other events.

Reply:
Haha, well yeah, i am being stingy on  amount of the components. But yeah, if we can have other things stuff into $supports helper, that will be great 👍

Reply:
And think about this:
```
window.AlpineMagicHelpers.Config = {
   supports: {
      *thing: *logic
   }
}}

where users can define their own thing and use it anywhere

Reply:
Yeah well the supports helper can work similar to screen, but separate. 

Also, I'd be fine having hundreds of helpers. We would just eventually have to upgrade the documentation. I'd like to start implementing jQuery features more so I can imagine having a lot.